File: leetCode/fukun/101-200/lc_155.js
/**
 * initialize your data structure here.
 */
var MinStack = function () {
    this.values = [];
    this.minValue = [];
};

/**
 * @param {number} x
 * @return {void}
 */
MinStack.prototype.push = function (x) {
    this.minValue.push(this.minValue[0] ? Math.min(this.top(), x) : x);
    return this.values.push(x);
};

/**
 * @return {void}
 */
MinStack.prototype.pop = function () {
    this.minValue.pop();
    return this.values.pop();
};

/**
 * @return {number}
 */
MinStack.prototype.top = function () {
    return this.values[this.values.length - 1];
};

/**
 * @return {number}
 */
MinStack.prototype.getMin = function () {
    return this.minValue[this.minValue.length - 1];
};

var obj = new MinStack();
console.log(obj.push(1));
console.log(obj.top());
console.log(obj.getMin());
console.log(obj.pop());

File: leetCode/XSS/lc_155.java
import java.util.Stack;

public class lc_155 {
    public static void main(String[] args) {
        //最小栈
    }
    static class MinStack {

        /** initialize your data structure here. */
        Stack<Integer> minStack;
        Stack<Integer> stack;
        public MinStack() {
            minStack=new Stack<>();
            stack=new Stack<>();
        }

        public void push(int x) {
            if(stack.isEmpty()||minStack.peek()>x)minStack.push(x);
            else
            {
                minStack.push(minStack.peek());
            }
            stack.push(x);
        }

        public void pop() {
            minStack.pop();
            stack.pop();
        }

        public int top() {
            return stack.peek();
        }

        public int getMin() {
            return minStack.peek();
        }
    }

}
